### PR TITLE
AppSortViewModelTest and ComponentSortViewModelTest

### PR DESCRIPTION
**Pull request by juliewww**
---
name: Pull request
about: Create a pull request
label: 'triage me'
---
Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Make sure to open a GitHub issue as a bug/feature request before writing your code! That way
  we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter
  pass (`./gradlew --init-script gradle/init.gradle.kts spotlessApply` to automatically apply
  formatting)
- [x] Appropriate docs were updated (if necessary)

Is this your first Pull Request?

- [ ] Run `./tools/setup.sh`
- [ ] Import the code formatting style as explained in [the setup script](/tools/setup.sh#L40).

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: core/model/src/main/kotlin/com/merxury/blocker/core/model/data/ComponentSortInfo.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package com.merxury.blocker.core.model.data
+
+import com.merxury.blocker.core.model.preference.ComponentShowPriority
+import com.merxury.blocker.core.model.preference.ComponentShowPriority.DISABLED_COMPONENTS_FIRST
+import com.merxury.blocker.core.model.preference.ComponentSorting
+import com.merxury.blocker.core.model.preference.ComponentSorting.COMPONENT_NAME
+import com.merxury.blocker.core.model.preference.SortingOrder
+import com.merxury.blocker.core.model.preference.SortingOrder.ASCENDING
+
+data class ComponentSortInfo(
```
Comment (lihenggui) [inline core/model/src/main/kotlin/com/merxury/blocker/core/model/data/ComponentSortInfo.kt:26]: Is that necessary to put this data class to the `model` module?

File: feature/sort/src/test/kotlin/com/merxury/blocker/feature/sort/ComponentSortViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        assertEquals(
+            Success(defaultUserData.toComponentSortInfo()),
+            viewModel.componentSortInfoUiState.value,
+        )
+
+        collectJob.cancel()
+    }
+
+    @Test
+    fun componentSortInfoUiState_updateComponentSorting() = runTest {
```
Comment (lihenggui) [inline feature/sort/src/test/kotlin/com/merxury/blocker/feature/sort/ComponentSortViewModelTest.kt:68]: Follow given_when_then format

File: core/model/src/main/kotlin/com/merxury/blocker/core/model/data/ComponentSortInfo.kt (hunk 1 above)
Comment (juliewww) [reply core/model/src/main/kotlin/com/merxury/blocker/core/model/data/ComponentSortInfo.kt:26]: Ok, move it back.

File: feature/sort/src/test/kotlin/com/merxury/blocker/feature/sort/ComponentSortViewModelTest.kt (hunk 2 above)
Comment (juliewww) [reply feature/sort/src/test/kotlin/com/merxury/blocker/feature/sort/ComponentSortViewModelTest.kt:68]: componentSortInfoUiState_whenUpdateComponentSorting_thenUpdateComponentSorting ?
I think it's not necessary.

File: core/model/src/main/kotlin/com/merxury/blocker/core/model/preference/UserPreferenceData.kt [hunk 3]
```diff
@@ -52,4 +54,16 @@ data class UserPreferenceData(
         darkThemeConfig = darkThemeConfig,
         useDynamicColor = useDynamicColor,
     )
+
+    fun toAppSortInfo() = AppSortInfo(
```
[the remark this answers is not in the document]
Comment (juliewww) [reply core/model/src/main/kotlin/com/merxury/blocker/core/model/preference/UserPreferenceData.kt:58]: ok

**Comment by juliewww** [comment]
Will resolve all comments in next PR.

**Pull request merged**